mcproto/types/nbt.py

```diff
@@ -342,6 +342,7 @@ def from_object(data: FromObjectType, schema: FromObjectSchema, name: str = "")
                 raise ValueError("Use a list or a dictionary in the schema to create a CompoundNBT or a ListNBT.")
             # Check if the data contains the name (if it is a dictionary)
             if isinstance(data, dict):
+                data = cast("Mapping[str, FromObjectType]", data)
```

Comment:
This should get inferred when `FromObjectType` is narrowed to `dict`, does it not?

Reply:
In my editor at least they appear as `Mapping[Unknown]` and `Sequence[Unknown]` and cause red squiggles later, but when I add the explicit cast nobody complains about an unnecessary cast...

I can remove them if they make the code less clear

Reply:
That's interesting, I assume you're using vscode? Then you're probbaly using pylance, which is just a pyright wrapper.

Pyright can mostly figure it out:

![image](https://github.com/py-mine/mcproto/assets/20902250/5d8e9b6b-f3de-4fdc-b230-cfb192293d60)

The only difference is that in this case, if the dict values are sequences or a mappings, they're unknown. This is because pyright can have some trouble with recursive types. It's actually an issue even after casing, it's just one level deeper:

![image](https://github.com/py-mine/mcproto/assets/20902250/82e1a19c-c1b8-40e2-a783-bfce2e174122)

But if you're using pyright, this should be the same for you, I'm not sure why you'd be getting `Mapping[Unknown]` there.